src/main/java/frc/joysticks/JoystickManager.java

```diff
@@ -0,0 +1,53 @@
+package frc.joysticks;
+
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Robot;
+import frc.JoystickBindings;
+import frc.utils.alerts.Alert;
+
+public class JoystickManager {
+
+	private static final SmartJoystick[] joysticks = new SmartJoystick[JoystickConstants.NUMBER_OF_JOYSTICK_PORTS];
+
+	private static void setBindSet(int port, BindSet bindSet, Robot robot) {
+		if (port < 0 || port >= JoystickConstants.NUMBER_OF_JOYSTICK_PORTS) {
+			new Alert(Alert.AlertType.ERROR, "you tried to create a joystick for a port that doesn't exist");
+			return;
+		} else if (joysticks[port] == null) {
+			createJoystick(port, robot);
+		}
+		joysticks[port].setBindSet(bindSet);
+	}
+
+	public static void cycleBindSet(int port, Robot robot) {
+		if (joysticks[port].getBindSet().getIndex() == BindSet.values().length - 1) {
+			setBindSet(port, BindSet.getBindSetByIndex(1), robot);
+		} else {
+			setBindSet(port, BindSet.getBindSetByIndex(joysticks[port].getBindSet().getIndex() + 1), robot);
+		}
+	}
+
+	private static void createJoystick(int port, Robot robot) {
+		joysticks[port] = new SmartJoystick(port, BindSet.NONE);
+		JoystickBindings.configureBindings(joysticks[port], robot);
+	}
+
+	private static void addOptions(SendableChooser<BindSet> chooser, int port, Robot robot) {
+		chooser.setDefaultOption("NONE", BindSet.NONE);
```

Comment:
```suggestion
		chooser.setDefaultOption(Bindset.NONE.name, BindSet.NONE);
```
magic string, if you change the name to something else you will have to change this too